### PR TITLE
COMMERCE-5939 Dataset Fixes + minor updates

### PR DESCRIPTION
**Pull request by FabioDiegoMastrorilli**
This PR concerns dataset displays, iframes and the way they communicate.

**COMMERCE-5939**
when a dataset got rendered within an "ostile" container with a `display: none` css property, `cellRef.current.getClientRects()` returned an empty array causing this code to break the component.

```
const {width} = cellRef.current.getClientRects()[0];
registerColumn(columnName, width, resizable);
```

I used an IntersectionObserver to avoid the problem. The callback gets invoked only the first time the component gets rendered and, eventually (if it is hidden), when it becomes visible, then it disconnects the Observer.

**COMMERCE-5968**
In order to let customers using Java extension points and allow them to render content using JSPs, we massively deal with iframes. Wherever we want some content to be rendered within a side panel we wrap it within a `side-panel-content` taglib.

I did some refactoring and I removed the old side panel implementation we had in commerce module since it's now part of the dataset in `frontend-taglib-clay`.

### CODE REVIEW

**Comment by FabioDiegoMastrorilli** [comment]
ci:test:sf

**Comment by jbalsas** [comment]
Hey @FabioDiegoMastrorilli, I'm having a hard time wrapping my head around this. Is there an easy way to test the issue? The LPS only mentions a "hostile" environment, but how do we go and reproduce it?

Since this is mostly removals in `commerce` modules and you still have the most knowledge about this I'd lean towards just forwarding this, but it would be good to be able to see how is this actually being used so we can do a better job at maintaining it and evolving it in the future.

@wincent, do you want to maybe just do a quick 30k feet review in case something triggers your spidey-sense?

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/dataset_display/DatasetDisplay.js [hunk 1]
```diff
@@ -45,9 +44,7 @@ function DatasetDisplay(props) {
 	const [views, updateViews] = useState(props.views);
 	const [loading, setLoading] = useState(false);
 	const [actionLoading, setActionLoading] = useState(false);
-	const [datasetDisplaySupportSidePanelId] = useState(
-		props.sidePanelId || 'support-side-panel-' + getRandomId()
-	);
+	const [datasetDisplaySupportSidePanelId] = useState(props.sidePanelId);
```
Comment (wincent) [inline modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/dataset_display/DatasetDisplay.js:47]: It is a little odd that you would use `useState` for something that you never change. If you just want to associate a value once with a component and never update it, `useRef` is probably the right tool for that (in fact, IIRC, if you look at the implementation of `useState` in React, you'll see that it is using refs under the covers to associate data with components).

File: modules/apps/commerce/commerce-frontend-taglib/src/main/resources/META-INF/resources/side_panel_content/index.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+		window.top.Liferay.fire(OPEN_MODAL_FROM_IFRAME, payload);
+	});
+
+	document.querySelector('body').classList.remove('open');
+
+	document
+		.querySelectorAll('.side-panel-iframe-close, .btn-cancel')
+		.forEach((trigger) => {
+			trigger.addEventListener('click', (e) => {
+				e.preventDefault();
```
Comment (wincent) [inline modules/apps/commerce/commerce-frontend-taglib/src/main/resources/META-INF/resources/side_panel_content/index.js:32]: [Best not to abbreviate](https://github.com/liferay/liferay-frontend-projects/blob/master/guidelines/general/naming.md#avoid-abbreviations) (ie. `e` → `event`).

**Review by wincent — COMMENTED** [review]
> do you want to maybe just do a quick 30k feet review in case something triggers your spidey-sense?

30k-foot review didn't turn anything up.

spiderman

I'd be fine with forwarding this.

File: modules/apps/commerce/commerce-frontend-taglib/src/main/resources/META-INF/resources/side_panel_content/index.js (hunk 2 above)
Comment (FabioDiegoMastrorilli) [reply modules/apps/commerce/commerce-frontend-taglib/src/main/resources/META-INF/resources/side_panel_content/index.js:32]: Omg, I'm really terrible with this rule. Sorry guys

**Comment by jbalsas** [comment]
> Hi Chema, this ticked explains the one of the possible scenarios where it breaks.

Thanks @FabioDiegoMastrorilli! I feel that's a much better ticket to solve than [COMMERCE-5949](https://issues.liferay.com/browse/COMMERCE-5949) since it has all the information... I'd recommend trying to prioritize the more detailed reports over the others. Will make it easier for everyone to track down why things are the way they are :)

If you're going to make any changes, would you mind renaming the commits to point to [COMMERCE-5939](https://issues.liferay.com/browse/COMMERCE-5939) instead and close the other one as duplicate?

**Comment by FabioDiegoMastrorilli** [comment]
Of course! no problem!

File: modules/apps/commerce/commerce-frontend-taglib/src/main/resources/META-INF/resources/side_panel_content/index.js (hunk 2 above)
Comment (wincent) [reply modules/apps/commerce/commerce-frontend-taglib/src/main/resources/META-INF/resources/side_panel_content/index.js:32]: I wouldn't worry too much about this. We can actually lint for the common cases (eg. `e`, `evt` and friends) and then you won't have to worry about it. I thought we might have an issue for that already but I couldn't find it, so I created [this one](https://github.com/liferay/liferay-frontend-projects/issues/437).

**Comment by FabioDiegoMastrorilli** [comment]
ci:forward

**Pull request closed without merging**